packages/block-library/src/search/index.php

```diff
@@ -66,14 +66,14 @@ function render_block_core_search( $attributes ) {
 
 	if ( $show_input ) {
 		$input_classes = array( 'wp-block-search__input' );
-		if ( $is_button_inside ) {
+		if ( ! $is_button_inside && ! empty( $border_color_classes ) ) {
 			$input_classes[] = $border_color_classes;
 		}
 		if ( ! empty( $typography_classes ) ) {
 			$input_classes[] = $typography_classes;
 		}
 		$input_markup = sprintf(
-			'<input type="search" id="%s" class="wp-block-search__input %s" name="s" value="%s" placeholder="%s" %s required />',
+			'<input type="search" id="%s" class="%s" name="s" value="%s" placeholder="%s" %s required />',
```

Comment:
Minor Fix: While I was working I noticed that the `wp-block-search__input` class is duplicated. This class is not needed here as [it was added first](https://github.com/WordPress/gutenberg/blob/2d4949606169ca88010ebcd23da720c5f9d015b8/packages/block-library/src/search/index.php#L68).